### PR TITLE
Allow debug flags to be set at TDS startup

### PR DESCRIPTION
**Pull request by lesserwhirls**
Set debug flags using -Dtds.debug.flags="flag1 flag2 ... flagN"
(example: -Dtds.debug.flags="H5iosp/read H5iosp/filePos")

### CODE REVIEW

**Comment by lesserwhirls** [comment]
Let's try this @cwardgar

**Comment by lesserwhirls** [comment]
I think this means we could remove all the manual set debug flag calls from our tests and clean-up the output by default...if a test fails, then we could turn on specific debug flags using the new property to get more info.

**Comment by dopplershift** [comment]
Are things smart enough that you could have the debug dumped to a log and then just cat that log when the test fails? Just wondering if there's a way to eliminate double-running of test.

**Comment by DennisHeimbigner** [comment]
If I read this correctly, the classes for which this would work
is hardwired into TdsInit. Correct? Is there any way to use
reflection to generalize this?

File: cdm/src/main/java/ucar/nc2/util/DebugFlagsImpl.java [hunk 1]
```diff
@@ -71,4 +71,9 @@ public void set(String flagName, boolean value) {
     map.put(flagName, value);
   }
 
+  public String getSetFlags() {
+    String flags = map.keySet().toString();
+    return (flags.contains("${tds.debugflags}")) ? "" : flags;
```
Comment (cwardgar) [inline cdm/src/main/java/ucar/nc2/util/DebugFlagsImpl.java]: What does this do?

Comment (lesserwhirls) [inline cdm/src/main/java/ucar/nc2/util/DebugFlagsImpl.java]: It's cruft from before I set the default property value in TdsContext. I'll update to remove it.

**Comment by lesserwhirls** [comment]
@DennisHeimbigner - I think it's possible to use reflection, but this was pretty straightforward. I think we'd still need some class info in `tdsInit` though. I think it would be a good maintenance thing to do, and might require some interface changes to make sure all calls were consistent. At this point, I'm just happy that end users will be able to turn on debug flags on their systems to report info back to us, rather than sending huge files back or running a custom built war with flags turned on.

**Comment by lesserwhirls** [comment]
@dopplershift - unfortunately, things are not that smart, although I am sure gradle would be able to handle the logic.

**Comment by cwardgar** [comment]
In addition to reflection, another way to generalize debug flag setting is to Springify `cdm`. Then, Spring could inject a `DebugFlagsImpl` instance into any object that needed it. But long term, I think the best way to handle this problem is to just write the debug messages to dedicated debug loggers. Then those messages could be enabled/disabled by editing the logger configuration file (log4j2.xml or similar).

Anyway, this looks like a good straightforward approach for now.

**Pull request merged**